### ISSUE TITLE
collaboration

### ISSUE THREAD

**Issue by frostworx** [JordanViknar/SimpleSteamTinker#6]
Hi there,

Creator and former dev of steamtinkerlaunch here.

First, I appreciate the idea of a total rewrite of steamtinkerlaunch (if I knew how fast and big it would grow I would have never started writing it in bash initially, which is the main reason why it was never that user-friendly as it could have been).

I wish you all the best with that project.

I no longer work on steamtinkerlaunch for a longer time and passed the project to sonic2kk,
who is still very busy with working on the project and unfortunately still seems mostly to work on it alone (as I did).
It might be a good idea to look for collaboration for both projects (maybe by opening an issue) and possibly work together on a merge.

**Comment by JordanViknar** [comment]
> Hi there,
> 
> Creator and former dev of steamtinkerlaunch here.
> 
> First, I appreciate the idea of a total rewrite of steamtinkerlaunch (if I knew how fast and big it would grow I would have never started writing it in bash initially, which is the main reason why it was never that user-friendly as it could have been).
> 
> I wish you all the best with that project.
> 
> I no longer work on steamtinkerlaunch for a longer time and passed the project to sonic2kk, who is still very busy with working on the project and unfortunately still seems mostly to work on it alone (as I did). It might be a good idea to look for collaboration for both projects (maybe by opening an issue) and possibly work together on a merge.

Oh wow, hello, glad to see my little project has gotten your attention.

In case you want to know, I intend to possibly drop this version of the project. While I'd love to keep Lua as the language for it, especially since it's extremely simple yet efficient, it has its limitations at the end of the day.
Mostly, in this case, LGI being not well maintained nowadays.

Python could have been ideal as replacement, but I feared the program's performance would eventually suffer, and I thought it'd be ideal if I could fit everything in the least amount of files possible this time.

I have been toying with Rust recently, originally for unrelated reasons. I eventually figured it wouldn't actually be a bad choice for a rewrite at all, but with one cost though : the code's simplicity would be... well Rust isn't exactly known to be that easy, especially compared to Lua.
But Rust has a much stronger community in regards to GTK than Lua though. The [awesome-gtk](https://github.com/valpackett/awesome-gtk) repo features **49** Rust projects, in comparison to **2** Lua projects, counting mine.
I have also been toying around with [Relm4](https://crates.io/crates/relm4) to simplify the code, but I'm not certain I'll stick with it yet.

So far, the rewrite is still experimental and extremely unfinished, with only the Steam library detection being done. However, it's even faster, but most of all, extremely resistant to errors so far.

> unfortunately still seems mostly to work on it alone (as I did)

Which is the part of the reason why I chose Lua originally. To make it extremely easy to contribute and encourage collaboration.

> It might be a good idea to look for collaboration for both projects (maybe by opening an issue) and possibly work together on a merge.

A direct collaboration might be considered when this or the rewrite is mature enough.
